src/Compare.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver;
+
+use Composer\Semver\Constraint\VersionConstraint;
+use InvalidArgumentException;
+
+class Compare
```

Comment:
Can you expand on your suggestion? What would be the drawback of allowing extending?